File: win-prog/ProgWinCS/Hello Windows Forms/TwoPaintHandlers/TwoPaintHandlers.cs
//-----------------------------------------------
// TwoPaintHandlers.cs © 2001 by Charles Petzold
//-----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class TwoPaintHandlers
{
     public static void Main()
     {
          Form form      = new Form();
          form.Text      = "Two Paint Handlers";
          form.BackColor = Color.White;
          form.Paint    += new PaintEventHandler(PaintHandler1);
          form.Paint    += new PaintEventHandler(PaintHandler2);

          Application.Run(form);
     }
     static void PaintHandler1(object objSender, PaintEventArgs pea)
     {
          Form     form = (Form)objSender;
          Graphics grfx = pea.Graphics;

          grfx.DrawString("First Paint Event Handler", form.Font, 
                          Brushes.Black, 0, 0);
     }
     static void PaintHandler2(object objSender, PaintEventArgs pea)
     {
          Form     form = (Form)objSender;
          Graphics grfx = pea.Graphics;

          grfx.DrawString("Second Paint Event Handler", form.Font, 
                          Brushes.Black, 0, 100);
     }
}

File: win-prog/ProgWinVB/Hello Windows Forms/TwoPaintHandlers/TwoPaintHandlers.vb
'-------------------------------------------------
' TwoPaintHandlers.vb (c) 2002 by Charles Petzold
'-------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Module TwoPaintHandlers
    Sub Main()
        Dim frm As New Form()
        frm.Text = "Two Paint Handlers"
        frm.BackColor = Color.White
        AddHandler frm.Paint, AddressOf PaintHandler1
        AddHandler frm.Paint, AddressOf PaintHandler2

        Application.Run(frm)
    End Sub

    Sub PaintHandler1(ByVal obj As Object, ByVal pea As PaintEventArgs)
        Dim frm As Form = DirectCast(obj, Form)
        Dim grfx As Graphics = pea.Graphics
        grfx.DrawString("First Paint Event Handler", frm.Font, _
                        Brushes.Black, 0, 0)
    End Sub

    Sub PaintHandler2(ByVal obj As Object, ByVal pea As PaintEventArgs)
        Dim frm As Form = DirectCast(obj, Form)
        Dim grfx As Graphics = pea.Graphics
        grfx.DrawString("Second Paint Event Handler", frm.Font, _
                        Brushes.Black, 0, 100)
    End Sub
End Module
